osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -33,6 +33,14 @@ public class ChannelManager : PollingComponent, IChannelPostTarget
         private readonly BindableList<Channel> availableChannels = new BindableList<Channel>();
         private readonly BindableList<Channel> joinedChannels = new BindableList<Channel>();
 
+        // Keeps a list of closed channels. More recently closed channels appear at higher indeces
+        private readonly BindableList<Channel> closedChannels = new BindableList<Channel>();
+
+        // For efficiency purposes, this constant bounds the number of closed channels we store.
+        // This number is somewhat arbitrary; future developers are free to modify it.
+        // Must be a positive number.
+        private const int closed_channels_max_size = 50;
```

Comment:
We actually have an existing utility collection that does this in this repository (see `LimitedCapacityQueue`). It would have to be moved out from its current namespace first, though, so unsure if worth - and even then maybe best left for a separate pull.